sklearn/neighbors/lof.py

```diff
@@ -250,26 +262,52 @@ def _predict(self, X=None):
         if X is not None:
             X = check_array(X, accept_sparse='csr')
             is_inlier = np.ones(X.shape[0], dtype=int)
-            is_inlier[self._decision_function(X) < 0] = -1
+            is_inlier[self.decision_function(X) < 0] = -1
         else:
             is_inlier = np.ones(self._fit_X.shape[0], dtype=int)
             is_inlier[self.negative_outlier_factor_ < self.offset_] = -1
 
         return is_inlier
 
-    def _decision_function(self, X):
+    def predict(self, X):
+        """Predict the labels (1 inlier, -1 outlier) of X according to LOF.
+
+        This method allows to generalize prediction to *new observations* (not
+        in the training set). Only available for novelty detection (when
+        novelty is set to True).
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            The query sample or samples to compute the Local Outlier Factor
+            w.r.t. to the training samples.
+
+        Returns
+        -------
+        is_inlier : array, shape (n_samples,)
+            Returns -1 for anomalies/outliers and +1 for inliers.
+        """
+
+        if self.novelty:
+            return self._predict(X)
+
+        msg = ('predict is not available when novelty=False, use fit_predict '
+               'if you want to predict on training data. Use novelty=True if '
+               'you want to use LOF for novelty detection and predict on new '
+               'unseen data.')
+        raise NotImplementedError(msg)
```

Comment:
ValueError

as I said elsewhere I think it more natural to finish a function with a return than a raise though